### PR TITLE
Update User Entity to include Gender property on SignUp use case.

### PR DESCRIPTION
**Pull request by nelegatti**
## Pull request type

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

SignUp logic is not allowing Gender property on User entity

Issue Link: N/A

## What is the new behavior?

Users can be created with Gender value. User's entity updated for match the mockups requirements.

## Other information

 - I take into account the 3 genders that are recommended, I put an enum since the mockup has the phrase "Select your gender", so I understand that it has to be a collection of values and is more maintainable, we can add any other Gender that the logic requires.
 - Readme updated

### CODE REVIEW

**Review by zk182 — APPROVED** [review]

File: src/database/factories/user.factory.ts [hunk 1]
```diff
@@ -2,18 +2,21 @@ import * as Faker from 'faker';
...
 import { User } from '@entities/user.entity';
 import { genSaltSync, hashSync } from 'bcrypt';
+import { Genders } from '@constants/enums/genders';
 
 define(User, (faker: typeof Faker) => {
   const firstName = faker.name.firstName();
   const lastName = faker.name.lastName();
   const email = faker.internet.email(firstName, lastName);
   const password = 'password123';
+  const randomGender = faker.random.arrayElement(Object.values(Genders));
```
Comment (EMazzaglia) [inline src/database/factories/user.factory.ts:12]:
```suggestion
  const gente = faker.random.arrayElement(Object.values(Genders));
```

To keep with the standard variable names.

File: src/database/factories/user.factory.ts [hunk 2]
```diff
@@ -2,18 +2,21 @@ import * as Faker from 'faker';
...
 
   const user = new User();
   user.firstName = firstName;
   user.lastName = lastName;
   user.email = email;
   user.password = hashSync(password, genSaltSync());
+  user.gender = randomGender;
```
Comment (EMazzaglia) [inline src/database/factories/user.factory.ts:19]:
```suggestion
  user.gender = gender;
```

File: src/entities/user.entity.ts [hunk 3]
```diff
@@ -30,5 +35,10 @@ export class User extends Base {
 
   @Column({ select: false })
   @IsNotEmpty()
+  @MinLength(6)
   password!: string;
+
+  @Column({ nullable: true })
+  @IsEnum(Genders)
+  gender?: Genders;
```
Comment (EMazzaglia) [inline src/entities/user.entity.ts:43]: Do you think that gender can be null? Is not a requirement of the signup?

**Review by EMazzaglia — COMMENTED** [review]
Hi Jesus,  I left some comments and change suggestions. Can you check them out? 
Thanks!

File: src/database/factories/user.factory.ts (hunk 1 above)
Comment (nelegatti) [reply src/database/factories/user.factory.ts:12]: Great idea Emi ! One question, do you mean const gender, not gente , right?

File: src/entities/user.entity.ts (hunk 3 above)
Comment (nelegatti) [reply src/entities/user.entity.ts:43]: The decorator @IsEnum(Genders) shouldn't allow null values. By making the column nullable I just followed what was already made for the rest of the user entity, like firstName, etc. Sounds good?

File: src/database/factories/user.factory.ts (hunk 2 above)
Comment (nelegatti) [reply src/database/factories/user.factory.ts]: it's done, thank you!

**Pull request merged**